portal/server/controllers/dashboard/filters/ui-selected-filters.test.js

```diff
@@ -0,0 +1,94 @@
+import {
+  generateSelectedFiltersObject,
+  selectedDashboardFilters,
+} from './ui-selected-filters';
+import CONTENT_STRINGS from '../../../content-strings';
+import CONSTANTS from '../../../constants';
+
+describe('controllers/dashboard/filters - ui-selected-filters', () => {
+  describe('generateSelectedFiltersObject', () => {
+    it('should return mapped object', () => {
+      const mockHeading = CONTENT_STRINGS.DASHBOARD_FILTERS.FILTER_HEADINGS.PRODUCT;
+      const mockFieldName = CONSTANTS.FIELD_NAMES.DEAL_TYPE;
+      const mockSubmittedFieldFilters = [
+        CONSTANTS.PRODUCT.GEF,
+        CONSTANTS.PRODUCT.BSS_EWCS,
+      ];
+
+      const result = generateSelectedFiltersObject(
+        mockHeading,
+        mockFieldName,
+        mockSubmittedFieldFilters,
+      );
+
+      const expectedFormattedFieldValue = (fieldValue) => fieldValue.replace(/[\s+/]/g, '-').replace('\'', '');
```

Comment:
what type of fields do we expect here?

Reply:
Good spot - i'll move this to a seperate function and add some comments/docs - it's used in a couple of places; one has a comment, others don't

It will format a field/filter value into a 'friendly selector', e.g these:
```
Further Maker's input required
Approved by UKEF (with conditions)
BSS/EWCS
```

will become:
```
Further-Makers-input-required
Approved-by-UKEF-with-conditions
BSS-EWCS
```
which are then used for data-cy attributes

The one in this file I think might be different to others, but they can all use the same one. Will improve 👍

Reply:
Moved this function into a helper file, added comments/docs. Both places using the same function